README.md

```diff
@@ -1,49 +1,146 @@
-Though the examples above can be evaluated based on their relative error compared to an expected
-correct value, many real-world applications have unknown "truth", and thus can't be evaluated in
-this way.
+Computational analysis has become an essential component of science. The software tools
+used bear the weight of the countless models, discoveries, and interventions based
+upon them. However, computers were never intended to be perfect, and one doesn't
+need to look very far to find examples of where conceptually simple operations
+may fail due to floating point arithmetic (e.g. ...
 
-This project aims to provide an environment in which the stability of programs can be evaluated,
-so that developers may evaluate the trustworthiness of their tools even in the absense of truth.
+```
+$ python3 -c "print(sum([0.001] * 1000))"
+1.0000000000000007
+```
 
+). While small issues like the one above are unlikely to cause catestrophic failures on
+their own, they can cascade over the course of an execution and it isn't uncommon to
+find a [plane that requires a power cycle every two months](https://www.theregister.com/2020/04/02/boeing_787_power_cycle_51_days_stale_data)
+or an [unstable series that consistently converges to the wrong solution](https://nbviewer.jupyter.org/github/verificarlo/fuzzy/blob/master/tutorial/evaluating_unstable_sequence.ipynb).
 
-## Usage
-
-*Coming soon*!
+*Fuzzy* allows you to study the stability or *trustworthiness* of tools and pipelines.
+You start by *instrumenting* libraries in your pipeline which do the bulk of numerical
+heavy lifting, then run your tool multiple times, and finally analyze the variability
+in your results. This project aims to provide an environment in which the stability of
+programs can be evaluated, reducing the overhead on both tool developers and consumers.
 
-## Supported Languages/Libraries
-
-- BLAS
-- LAPACK
-- Python3.6.5
-- Numpy (Python3.6.5)
-- Pytorch (Python3)
-- Libmath
 
+## Usage
 
-## Build Instructions
+#### Building & booting the environment
+You can get started with *Fuzzy* quite simply, just launch a
+[Docker](https://www.docker.com/) container as follows:
+
+```
+docker run -ti verificarlo/fuzzy
+```
+
+If you're on a shared system and would prefer to use [Singularity](https://sylabs.io/),
+that's no problem, just convert the container using the appropriate method for
+your system ([e.g.](https://docs.computecanada.ca/wiki/Singularity#Creating_images)).
+
+If you would like to build the environment locally on your system, look at the
+Dockerfiles in `docker/base/` to see how installation was performed.
+
+#### Adding your software
+If your code is written in, say, Python, then you can simply mount your code to
+this environment, run it using the contained interpreter, and experince the *:fuzz:*.
+If your code loads shared libraries, such as `libmath` or `lapack`, make sure that
```

Comment:
We should refer to libmath as "GNU C Library" and link to https://www.gnu.org/software/libc/manual/html_node/Mathematics.html

Reply:
Why refering to GNU C lib? People can use whatever libmath they want

Reply:
fair point